tests/frontier/scenarios/test_scenarios.py

```diff
@@ -0,0 +1,261 @@
+        program_logs,
+        # 00 - 20
+        program_all_frontier_opcodes,
+        # 30
+        program_address,
+        program_balance,
+        program_origin,
+        program_caller,
+        program_callvalue,
+        program_calldataload,
+        program_calldatasize,
+        program_calldatacopy,
+        program_codecopy_codesize,
+        program_gasprice,
+        program_ext_codecopy_codesize,
+        program_returndatasize,
+        program_returndatacopy,
+        program_extcodehash,
+        # 40
+        program_blockhash,  # fails in state mode in py-t8n
+        program_coinbase,
+        program_timestamp,
+        program_number,
+        program_difficulty_randao,
+        program_gaslimit,
+        program_chainid,
+        program_selfbalance,
+        program_basefee,
+        program_blobhash,
+        program_blobbasefee,
+        program_tload,
+        program_mcopy,
+        program_push0,
+        program_suicide,
+    ],
+)
+def test_scenarios(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    fork: Fork,
+    expected_result: ProgramResult,
+    scenarios,
+):
+    """
+    Test given operation in different scenarios
+    Verify that it's return value equal to expected result on every scenario,
+    that is valid for the given fork.
+
+    Note: Don't use pytest parametrize for scenario production, because scenarios will be complex
+    Generate one test file for [each operation] * [each scenario] to save space
+    As well as operations will be complex too
+    """
+    # Skip disabled scenarios in debug mode
+    if len(scenarios) == 0:
+        return
+
+    tx_env = Environment()
+    post = Storage()
+    result_slot = post.store_next(1, hint="runner_result")
+
+    tx_origin: Address = pre.fund_eoa()
+    tx_gasprice: int = 10
+    exec_env = ExecutionEnvironment(
+        fork=fork,
+        origin=tx_origin,
+        gasprice=tx_gasprice,
+        timestamp=tx_env.timestamp,
+        number=tx_env.number,
+        gaslimit=tx_env.gas_limit,
+        coinbase=tx_env.fee_recipient,
+    )
+
+    def make_result(scenario: Scenario) -> int:
+        """Make Scenario post result."""
+        if scenario.halts:
+            return post.store_next(0, hint=scenario.name)
+        else:
+            return post.store_next(
+                translate_result(expected_result, scenario.env, exec_env), hint=scenario.name
+            )
+
+    runner_contract = pre.deploy_contract(
+        code=sum(
+            Op.MSTORE(0, 0)
+            + Op.CALL(10000000, scenario.code, 0, 0, 0, 0, 32)
+            + Op.SSTORE(make_result(scenario), Op.MLOAD(0))
+            for scenario in scenarios
+        )
+        + Op.SSTORE(result_slot, 1),
+        storage={
+            result_slot: 0xFFFF,
+        },
+    )
+
+    tx = Transaction(
+        sender=tx_origin,
+        gas_limit=500_000_000,
+        gas_price=tx_gasprice,
+        to=runner_contract,
+        data=b"0x11223344",
```

Comment:
I think this is what you meant:
```suggestion
        data=b"\x11\x22\x33\x44",
```
The current one creates a byte array with the ascii equivalent of the "0x11223344" string.

Reply:
is there a handy constructor like  BYTES("0x11223344")  ?

Reply:
`bytes.fromhex()`, example here: https://github.com/ethereum/execution-spec-tests/pull/1067#discussion_r1910121660.

Reply:
I see here is confusion
`bytes.fromhex("11223344"),`
prefix 0x is not allowed, can mistake with dec